src/decoders/arw.rs

```diff
@@ -73,7 +73,7 @@ impl<'a> Decoder for ArwDecoder<'a> {
 }
 
 impl<'a> ArwDecoder<'a> {
-  fn image_a100(&self, camera: &Camera) -> Result<RawImage,String> {
```

Comment:
It's very hard to make rust leak memory. In this case since orientation is Copy it won't even reference the original one. Even if it wasn't it would just mean ok_image would need to clone. Just leave all those signatures unchanged and the compiler will tell you if you need to do anything else.